framework/src/actions/AutoCheckpointAction.C

```diff
@@ -0,0 +1,31 @@
+#include "AutoCheckpointAction.h"
+#include "FEProblem.h"
+#include "Checkpoint.h"
+registerMooseAction("MooseApp", AutoCheckpointAction, "auto_checkpoint_action");
+
+InputParameters
+AutoCheckpointAction::validParams()
+{
+  InputParameters params = Action::validParams();
+
+  return params;
+}
+
+AutoCheckpointAction::AutoCheckpointAction(const InputParameters & params) : Action(params) {}
+
+void
+AutoCheckpointAction::act()
+{
+  if (_app.isUltimateMaster())
+  {
+    if (!_app.getOutputWarehouse().getOutputs<Checkpoint>().empty())
+      // if there's already a checkpoint object, we don't need to worry about creating a new
+      // checkpoint
+      return;
+
+    auto cp_params = _factory.getValidParams("Checkpoint");
+    cp_params.setParameters("should_output", false);
+    cp_params.setParameters("is_autosave", true);
```

Comment:
You don't need too Booleans to enable one behavior. This leads to some weird behavior since you now have four combinations of parameters, two of only which are defined. At the very least only expose one parameter here, even if you end up using two for convenience inside the object.

Reply:
So, the purpose of this is that since we are using the one Checkpoint class, if you don't have a flag that checks "is this the autosave checkpoint", it will stop outputting a checkpoint created by `Outputs/checkpoint=true` when MOOSE receives the SIGUSR1. I could just make should_output a private boolean with a setter, and not a param, but the two bools are there for a real reason. If we also don't care about the aforementioned behavior, then is_autosave can be removed.

Reply:
You may need two Booleans, but you only need to expose one.